proximal/examples/test_deconv_sv_psf.py

```diff
@@ -0,0 +1,320 @@
+    scale_factor = np.sum(psf_modes, axis=(1, 2)).max() * n_psf * weights.max()
+    blur_op = LinOpFactory(im_shape,
+                           im_shape,
+                           forward,
+                           adjoint,
+                           norm_bound=scale_factor)
+
+    return blur_op
+
+
+def NagyAndOLeary1998(x,
+                      im,
+                      psf_modes,
+                      weights,
+                      mu=0.01,
+                      n_psf=15) -> px.lin_ops.black_box.BlackBox:
+    ''' Convolve first, then apply weights.
+
+        According to Eq. 23 of the review article (link below), there exists a
+        non-equivalent image interpolation formulation introduced by Nagy and
+        O'Leary.
+         
+        Reference: Denis, Thiebaut, Soulez et al. 2015,
+        https://doi.org/10.1007/s11263-015-0817-x
+    '''
+    blur_op = planewise_conv(psf_modes, im.shape, n_psf)
+    im_blur = blur_op(x)
+
+    im_stack = np.empty(weights.shape, dtype=np.float32, order='F')
+    im_stack[:] = im[np.newaxis, ...]
+
+    # The weights are manually "absorbed" into the L2-norm, which has a fast
+    # elemwise proximal operator.
+    #
+    # TODO: Crop the im_stack and weight to double of image grid size; elminiate
+    # redundant 2D convolution with zero weights.
+    data_term = px.weighted_sum_squares(im_blur - im_stack, weights)
+
+    im_grad = px.grad(x)
+    grad_term = mu * px.norm1(im_grad) + (1 - mu) * px.sum_squares(im_grad)
+    return data_term + grad_term  #+ px.nonneg(x)
+
+
+def FlickerAndRigaut2005(x,
+                         im,
+                         psf_modes,
+                         weights,
+                         mu=0.01,
+                         alpha=0.9,
+                         n_psf=15,
+                         using_blackbox=False):
+    ''' Weight, then convolve.
+
+        According to Eq. 22 of the review article (link below), PSF
+        interpolation technique preserves PSF symmetry and normalization with
+        minimal sacrifice of the computation complexity.
+         
+        Reference: Denis, Thiebaut, Soulez et al. 2015,
+        https://doi.org/10.1007/s11263-015-0817-x
+    '''
+    if using_blackbox:
+        blur_op = weighted_planewise_conv(psf_modes, weights, im.shape, n_psf)
+        im_blur = blur_op(x)
+    else:
+        im_blur = px.sum([
+            px.conv(psf_modes[i], px.mul_elemwise(weights[i], x))
+            for i in range(n_psf)
+        ])
+
+    # The weights are manually "absorbed" into the L2-norm, which has a fast
+    # elemwise proximal operator.
+    #
+    # TODO: Crop the im_stack and weight to double of image grid size; elminiate
+    # redundant 2D convolution with zero weights.
+    data_term = px.sum_squares(im_blur - im)
+
+    im_grad = px.grad(x)
+    grad_term = mu * alpha * n_psf * px.group_norm1(im_grad, group_dims=[
+        2
+    ]) + mu * (1 - alpha) * n_psf * px.sum_squares(im_grad)
```

Comment:
Decoupled alpha and mu: `alpha` is the relative strengths between TV-L1 and TV-L2; `mu` is the relative `grad_term` strength to the `data_term`.